### ISSUE TITLE
Scheduled action not executed when state of person is unknown

### ISSUE THREAD

**Issue by pomeloy** [nielsfaber/scheduler-component#263]
### Checklist
- [x] I checked for similar existing issues (both open and closed) before posting.
- [x] I am running the latest version v3.2.14.

### Expected behavior
Scheduled action takes place

### Actual behavior
Scheduled action does not take place
Logs empty

### Steps to Reproduce

- Create any schedule, conditions should include a person (current state "unknown")
  - either as one of several conditions checking whether at least one of several person entities has the state "home" (wherein at least one of the other persons is home)
  - or as singular condition ("person NOT at home")

Actions never get executed whenever a person with unknown state is included in one of the conditions, whereas native automations work flawlessly, e.g. 

```
condition: not
conditions:
  - condition: state
    entity_id: person.person
    state: home
```

**Comment by pomeloy** [comment]
Still an issue.

**Comment by nielsfaber** [comment]
Sorry for the late response.
Scheduler uses its own [implementation ](https://github.com/nielsfaber/scheduler-component/blob/5cf84e70aae93a42f8f4d7d0a07be3a674d31ee3/custom_components/scheduler/actions.py#L173-L180)for evaluating conditions, which is very basic.
There is no code present to handle cases where a state is `unknown` or `unavailable`.
So in case you create a condition `person IS home`  any other state than exactly 'home' would cause the conditions to fail.
Likewise in case you create a condition `person IS NOT home`  any other state than exactly 'home' would cause the conditions to pass, including `unknown`.

Could you elaborate which behaviour you would expect here instead?
Pointing to the HA implementation is not very helpful since I cannot find any documentation of how cases with unknown/unavailable states are treated.

**Comment by pomeloy** [comment]
> So in case you create a condition person IS home any other state than exactly 'home' would cause the conditions to fail.

This is not what I have encountered. The condition fails whenever a sub-condition checks a person with the state unknown. 

What I expected: one unknown state should not fail the entire OR condition if at least one other person IS home

> Likewise in case you create a condition person IS NOT home any other state than exactly 'home' would cause the conditions to pass, including unknown.

This again is not what I encountered. The condition does not pass ‘IS NOT home’ whenever the state is unknown, compared to working as expected when checking a known state. 

What I expected: a check for person IS NOT home should pass whenever their state is unknown.
